proto/interchain_security/ccv/consumer/v1/consumer.proto

```diff
@@ -91,3 +92,9 @@ message ConsumerPackets {
   repeated ConsumerPacket list = 1
   [ (gogoproto.nullable) = false ];
 }
+
+// MaturingVSCPacket contains the maturing time of a received VSCPacket
```

Comment:
Prob worth noting that if this time is being stored on the consumer at all, the maturity time is in the future. Otherwise a VSCMatured packet is sent from consumer -> provider.

Is my understanding correct?

Reply:
Almost. This is set either in `InitGenesis` on a consumer restart in which case the maturity time could be in the past, which would result in `VSCMaturedPackets` sent in the first `EndBlock`. 

Most likely though, the maturity time is set in `OnRecvVSCPacket`, i.e., 
```golang
maturityTime := ctx.BlockTime().Add(k.GetUnbondingPeriod(ctx))
k.SetPacketMaturityTime(ctx, newChanges.ValsetUpdateId, maturityTime)
```